docs/en/examples/elementalarea_with_dataobject.md

```diff
@@ -0,0 +1,111 @@
+# Using ElementalArea with DataObject
+
+## Creating Model and adding ElementalArea
+
+In this example the class `BlogPost` will be created, it will extend the `DataObject` class. Also a new `BlogPost` Admin section will be created in the Admin panel where CMS users can manage the `BlogPost` objects.
+
+Let's look at an example:
+
+**app/src/Admins/BlogPostsAdmin.php**
+
+```php
+<?php
+
+namespace App\Admins;
+
+use App\Models\BlogPost;
+use SilverStripe\Admin\ModelAdmin;
+
+class BlogPostsAdmin extends ModelAdmin
+{
+    private static string $url_segment = 'blog-posts-admin';
+
+    private static string $menu_title = 'Blog Posts';
+
+    private static string $menu_icon_class = 'font-icon-block-banner';
+
+    private static array $managed_models = [
+        BlogPost::class,
+    ];
+}
+```
+
+`$managed_models` - this static variable connects `BlogPostAdmin` with `BlogPost`, which will be created in the following code.\
+When you create a `DataObject` you should define the connection between your `DataObject` and `ElementalArea` by adding `$has_one` relationship.\
+In this example `BlogPost` will have one `ElementalArea`.
+
+**app/src/Models/BlogPost.php**
+
+```php
+<?php
+
+namespace App\Models;
+
+use App\Admins\BlogPostsAdmin;
+use DNADesign\Elemental\Models\ElementalArea;
+use SilverStripe\Control\Controller;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Versioned\Versioned;
+
+class BlogPost extends DataObject
+{
+    private static string $table_name = 'BlogPost';
+
+    private static $has_one = [
+        'ElementalArea' => ElementalArea::class,
+    ];
+
+    private static $owns = ['ElementalArea'];
+
+    // ...
+}
+ ```
+
+If you are using `ElementalArea` together with `DataObject`, it is important to define a following method in the class.\
+The `CMSEditLink()` method should be defined to generate a link for opening the editing section of the particular `BlogPost`.\
+`BaseElement` uses a link, that was created in `CMSEditLink()` method in `BlogPost` class, to generate new links for opening the editing section of the particular element.
```

Comment:
```suggestion
If you are using `ElementalArea` together with `DataObject`, it is important to define the `CMSEditLink()` method in the class.\
`BaseElement::CMSEditLink()` relies on a valid CMS link being available in the parent `DataObject` - in this case, `BlogPost`. It is used to navigate directly to the editing section of the particular element (for example in the "Content blocks in use" report).

Note that for nested `GridField`s this method can get more complicated. Similarly, if your class is used in multiple admins, you will have to choose one to be the canonical admin section for the purposes of this method. This example only shows the simplest case, where the `BlogPost` class is used directly in `BlogPostsAdmin`.
```
The language here was a bit confusing - I _think_ this is what you were trying to say, but if I got it wrong just let me know and we can figure out a clear way to communicate the correct message.

I've also added a note here about some edge cases developers may encounter which aren't covered in this example.

Reply:
I think we don't need this -> (for example in the "Content blocks in use" report). We don't have any code examples related to "Content blocks in use"

Reply:
Fair enough. My intention was to give context to the reason `CMSEditLink()` is needed on the `DataObject` subclass, but if you don't think that part's needed I'm okay leaving it out.